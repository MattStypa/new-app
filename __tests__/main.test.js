const childProcess = require('child_process');

test('missing project', () => {
  const [stdout, stderr] = run();
  expect(stdout).toMatchSnapshot();
  expect(stderr).toMatchSnapshot();
});

test('missing destination', () => {
  const [stdout, stderr] = run('project');
  expect(stdout).toMatchSnapshot();
  expect(stderr).toMatchSnapshot();
});

function run(args = '') {
  try {
    const result = childProcess.execSync(`node main.js ${args}`, { stdio: 'pipe' });
    return [result.stdout.toString(), result.stderr.toString()];
  } catch (err) {
    return [err.stdout.toString(), err.stderr.toString()];
  }
}
