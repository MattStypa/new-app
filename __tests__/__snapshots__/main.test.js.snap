// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`missing destination 1`] = `
"  
   [1;37mnew-app[0m 0.2.2
  
  
   Usage:
   [1;37mnpx new-app[0m [1;35m<project> <directory>[0m
  
   For help resolving this problem please visit:
   [1;37mhttps://github.com/MattStypa/new-app/issues[0m
  
"
`;

exports[`missing destination 2`] = `
"   [1;31mDestination directory is required.[0m
"
`;

exports[`missing project 1`] = `
"  
   [1;37mnew-app[0m 0.2.2
  
  
   Usage:
   [1;37mnpx new-app[0m [1;35m<project> <directory>[0m
  
   For help resolving this problem please visit:
   [1;37mhttps://github.com/MattStypa/new-app/issues[0m
  
"
`;

exports[`missing project 2`] = `
"   [1;31mProject name is required.[0m
"
`;
